### PR TITLE
track recording amplitude and use to generate attachment preview

### PR DESCRIPTION
**Pull request by mmaxim**
The service has a special RPC to generate an audio preview if we already know the amplitudes of the recording. Use it to override whatever other preview the service will try to create from the audio file.

### CODE REVIEW

File: shared/chat/conversation/input-area/normal/platform-input.native.tsx [hunk 1]
```diff
@@ -112,6 +115,14 @@ class _PlatformInput extends PureComponent<PlatformInputPropsInternal, State> {
     }
   }
 
+  _enableAudioRecording = () => {
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/platform-input.native.tsx]: lets start to use `private` and not use `_` prefixes

File: shared/chat/conversation/input-area/audio.tsx [hunk 2]
```diff
@@ -15,7 +15,9 @@ export const AudioSend = (props: Props) => {
   // dispatch
   const dispatch = Container.useDispatch()
   const onCancel = () => {
-    dispatch(Chat2Gen.createStopAudioRecording({conversationIDKey, stopType: Types.AudioStopType.CANCEL}))
+    dispatch(
+      Chat2Gen.createStopAudioRecording({amps: [], conversationIDKey, stopType: Types.AudioStopType.CANCEL})
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/audio.tsx]: should make amps optional in the json

File: shared/chat/conversation/normal/index.native.tsx [hunk 3]
```diff
@@ -28,40 +27,34 @@ const Offline = () => (
...
 
-class Conversation extends React.PureComponent<Props> {
-  render() {
-    return (
-      <Box2 direction="vertical" fullWidth={true} fullHeight={true}>
-        {this.props.threadLoadedOffline && <Offline />}
-        <HeaderArea
-          onToggleInfoPanel={this.props.onToggleInfoPanel}
-          conversationIDKey={this.props.conversationIDKey}
+const Conversation = (props: Props) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/normal/index.native.tsx]: if you're going from pure to a functional component we should eval if it still needs to be pure or not, if so use `React.memo`

File: shared/chat/conversation/input-area/normal/platform-input.native.tsx [hunk 4]
```diff
@@ -34,22 +34,23 @@ class _PlatformInput extends PureComponent<PlatformInputPropsInternal, State> {
   _whichMenu?: menuType
   state = {hasText: false}
   _ampTracker = new AmpTracker(60)
+  _audioDragY = new Kb.NativeAnimated.Value(0)
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/platform-input.native.tsx]: this stuff can be private also

File: shared/chat/conversation/input-area/normal/platform-input.native.tsx [hunk 5]
```diff
@@ -337,19 +344,44 @@ const Action = React.memo((props: ActionProps) => {
...
 
-const maxAudioDrift = -20
+const maxCancelDrift = -20
+const maxLockDrift = -70
 
 const AudioStarter = (props: AudioStarterProps) => {
   let longPressTimer
+  const locked = React.useRef<boolean>(false)
+  React.useEffect(() => {
+    if (locked.current && !props.locked) {
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/platform-input.native.tsx]: you should do `const {locked} = props` so you don't close over props

Comment (mmaxim) [reply shared/chat/conversation/input-area/normal/platform-input.native.tsx]: Why? There are like 5 props.

**Pull request merged**